Models.py

```diff
@@ -185,6 +223,15 @@ def is_homogeneous(arr):
         plt.show()
 
     def load(self) -> Optional[Self]:
+        """
+        This method will load the model using the tensorflow load method.
+        
+        Args:
+            None
+
+        Returns:
+            self.model (Optional[Self]]): The model that was loaded
```

Comment:
It is not necessarily loaded. That is why it can return None.
How about something like:

Returns:
        None: If no model is loaded.
        model: The saved model if it is successfully saved.